plasmapy/diagnostics/tests/test_langmuir.py

```diff
@@ -1,40 +1,86 @@
-"""Tests for functions that use Langmuir analysis functions."""
+# coding=utf-8
+"""Tests for Langmuir probe analysis functions."""
 
 import numpy as np
+from astropy import units as u
+import pytest
+import plasmapy.constants as const
 
-from plasmapy.diagnostics.langmuir import (swept_probe_analysis,
-                                           obtain_EEDF,
-                                           obtain_EEPF)
-
-
-class Test_swept_probe_analysis():
-    def test_no_signal(self):
-        """
-        Checks whether an error is raised when zero signal is passed to the
-        function.
-        """
-        errStr = ("Performing a swept probe analysis should return an error "
-                  "when zero signal is passed, but doesn't.")
-        assert True, errStr
-
-
-class Test_obtain_EEDF():
-    def test_no_signal(self):
-        """
-        Checks whether an error is raised when zero signal is passed to the
-        function.
-        """
-        errStr = ("Attempting to obtain the EEDF should return an error when "
-                  "zero signal is passed, but doesn't.")
-        assert True, errStr
-
-
-class Test_obtain_EEPF():
-    def test_no_signal(self):
-        """
-        Checks whether an error is raised when zero signal is passed to the
-        function.
-        """
-        errStr = ("Attempting to obtain the EEPF should return an error when "
-                  "zero signal is passed, but doesn't.")
-        assert True, errStr
+from plasmapy.diagnostics.langmuir import (Characteristic,
+                                           swept_probe_analysis)
+
+N = 30  # length of test characteristic
+
+I_arr = np.random.rand(N) * u.A
+U_arr = np.random.rand(N) * u.V
+class Test__characteristic_errors:
+    r"""Test the Characteristic class constructor in langmuir.py"""
+    N = 30  # length of test characteristic
+    U_infarr = np.append(np.random.rand(N - 1), np.inf) * u.V
+    I_infarr = np.append(np.random.rand(N - 1), np.inf) * u.A
+
+    def test_infinite_I(self):
+        with pytest.raises(ValueError):
+            Characteristic(U_arr, self.I_infarr)
+
+    def test_infinite_U(self):
+        with pytest.raises(ValueError):
+            Characteristic(self.U_infarr, I_arr)
+
+
+@pytest.fixture
+def characteristic():
+    return Characteristic(U_arr, I_arr)
+
+@pytest.fixture
+def characteristic_simulated():
+    r""""Simulated characteristic check below (unfinished)"""
+    T_e_sim = 1 * u.eV
+    n_e_sim = 10**18 * u.m**-3
```

Comment:
`1e18` instead of `10 ** 18` so we have one less operator :smile:

Reply:
OPTIMIZATION

Reply:
:rofl: code is gonna be too fast now, must slow it down!